security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -322,12 +325,19 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 				"error details %s\n", conIDresourceNamePrefix, discReq.Node.Id, firstRequestFlag, discReq.VersionInfo,
 				discReq.ErrorDetail)
 
-			// In ingress gateway agent mode, if the first SDS request is received but kubernetes secret is not ready,
-			// wait for secret before sending SDS response. If a kubernetes secret was deleted by operator, wait
-			// for a new kubernetes secret before sending SDS response.
-			if s.st.ShouldWaitForIngressGatewaySecret(conID, resourceName, token) {
+			// In ingress gateway agent mode, if the first SDS request is received but Ingress gateway secret which is
+			// provisioned as kubernetes secret is not ready, wait for secret before sending SDS response.
+			// If a kubernetes secret was deleted by operator, wait for a new kubernetes secret before sending SDS response.
+			// If workload uses file mounted certs i.e. "FILE_MOUNTED_CERTS" is set to true, workdload loads certificates from
+			// mounted certificate paths and it does not depend on the presence of ingress gateway secret so
+			// we should skip waiting for it in that mode.
+			// File mounted certs for gateways is used in scenarios where an existing PKI infrastuctures delivers certificates
+			// to pods/VMs via files.
+			if s.st.ShouldWaitForIngressGatewaySecret(conID, resourceName, token, s.fileMountedCerts) {
 				sdsServiceLog.Warnf("%s waiting for ingress gateway secret for proxy %q\n", conIDresourceNamePrefix, discReq.Node.Id)
 				continue
+			} else {
+				sdsServiceLog.Infof("Workload is using file mounted certificates. Skipping waiting for ingress gateway secret")
```

Comment:
This log is not correct. There can be multiple reasons ShouldWaitForIngressGatewaySecret returns false.

Reply:
That is a great point. I will change the log